### PR TITLE
Added support for independent centering and scaling of X and  Y

### PR DESCRIPTION
**Pull request by Sm00thix**
It is now possible to center and scale X and Y independently. It is also possible to scale without prior centering.

### CODE REVIEW

**Comment by Sm00thix** [comment]
Hi @parmentelat ,
Sorry to bother you again with some changes to the code base. I just want to make sure you agree with the changes before I merge them to main as you are currently reviewing this repo over at [JOSS](https://github.com/openjournals/joss-reviews/issues/6533).

**The motivation**: After using the ikpls package myself and talking with some of my peers, I discovered the usefulness of being able to center and scale X and Y independently of eachother and also the ability to apply scaling without previous centering.

**The changes**: For all PLS algorithms, I replaced the `center` and `scale` parameters with the new `center_X`, `center_Y`, `scale_X`, and `scale_Y` parameters. I also modified the code so that it is possible to scale without centering. I modified the existing tests to account for this. I also added a new test that checks for equivalency between all algorithms for all 2^4=16 combinations of `center_X`, `center_Y`, `scale_X`, and `scale_Y`. I also updated the paper/time_pls.py script to account for these changes. Finally, I updated the README and all relevant documentation.

Will you allow me to merge this pull request to the main branch?

File: README.md [hunk 1]
```diff
@@ -43,12 +43,12 @@ cross-validation splits exceed 3. The fast cross-validation algorithms
 correctly handle (column-wise) centering and scaling of the X and Y
 input matrices using training set means and standard deviations to avoid
 data leakage from the validation set. This centering and scaling can be
-enabled by setting the center parameter to True and the scale parameter
-to True, respectively. The fast cross-validation algorithms correctly
-handle row-wise preprocessing such as (row-wise) centering and scaling
-of the X and Y input matrices, convolution, or other preprocessing.
-Row-wise preprocessing can safely be applied before passing the data to
-the fast cross-validation algorithms.
+enabled or disabled independently from eachother and for X and Y by setting
```
Comment (parmentelat) [inline README.md:46]: eachother: each other

**Comment by parmentelat** [comment]
Hi @Sm00thix 

as far as I can tell this change goes in the right direction, since it allows to do more than the previous API
so please go ahead with this PR :)

-- Thierry

**Comment by Sm00thix** [comment]
> Hi @Sm00thix
> 
> as far as I can tell this change goes in the right direction, since it allows to do more than the previous API so please go ahead with this PR :)
> 
> -- Thierry

This is correct. All of the previous functionalities are retained and new ones are added :)
Thank you for the feedback. I will merge the PR.

-- Ole

**Pull request merged**